### PR TITLE
Use Socket.tcp instead of TCPSocket

### PR DESCRIPTION
**Pull request by carlhoerberg**
Socket.tcp has a connect_timeout parameter, so that Timeout#timeout
isn't required. Socket.tcp with connect_timeout was introduced with 
Ruby 2.0.0 (Ruby 1.9 is EOL so it might be acceptable). 

This is a discussion point rather than a mergeable PR, related to #307.

### CODE REVIEW

File: lib/bunny/cruby/socket.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ module Bunny
   #
   # Heavily inspired by Dalli by Mike Perham.
   # @private
-  class Socket < TCPSocket
+  module Socket
```
Comment (michaelklishin) [inline lib/bunny/cruby/socket.rb]: Why is this now a module?

**Comment by michaelklishin** [comment]
I'm fine with requiring Ruby 2.0 from now on. The change looks OK. Please remove Ruby 1.9 from CI and update supported Ruby versions everywhere.

**Comment by carlhoerberg** [comment]
Remove a lot of cruby 1.8/1.9 specific stuff, updated travis, the readme etc.

**Comment by michaelklishin** [comment]
This probably can be Bunny 2.0. Not that we have any major API changes…

**Pull request merged**